tt-train/sources/ttml/core/tt_tensor_utils.cpp

```diff
@@ -181,13 +184,37 @@ tt::tt_metal::Tensor ones(const ttnn::Shape& shape, ttnn::distributed::MeshDevic
     return core::full(shape, 1.F, device, dtype);
 }
 
+template <>
+[[nodiscard]] tt::tt_metal::Tensor from_xtensors_to_host<float, DataType::BFLOAT16>(
+    const std::vector<xt::xarray<float>>& buffers, const std::unordered_map<std::string, std::string>& config) {
```

Comment:
Do you need to pass around `config` as a map of strings? IIUC this was just a convenience to pass some metadata from Python down to C++, so I think we can get away with the type `DistributedTensorConfig`

Reply:
Currently I reimplemented the same approach but yes I am planning to return DistributedTensorConfig directly.